File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int real = 0 ;
        int zero = 0 ;
        Map <Integer,Boolean> num = new HashMap<>();
        
        for(int i = 0 ; i < lottos.length ; i++){
            if(lottos[i] == 0){
                zero++;
                continue;
            }
            num.put(lottos[i],true);
        }
        
        for(int i = 0 ; i < win_nums.length ; i++){
            if(num.containsKey(win_nums[i])){
                real++;
            }
        }
        int maxnum = Math.min(6,7 - (real+zero));
        int minnum = Math.min(6,7 - real);
        
        int[] answer = {maxnum,minnum};
        
        return answer;
    }
}

File: C#/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cs
using System;
using System.Linq;
public class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
int[] answer = new int[2];
            int fNum = 0;
            int sNum = 0;
            foreach(int i in lottos)
            {
                if (i == 0) { sNum++; continue; }
                if (win_nums.Contains(i)) fNum++;
            }
            sNum += fNum;
            if (sNum > 1) answer[0] = 7 - sNum;
            else answer[0] = 6;

            if (fNum > 1) answer[1] = 7 - fNum;
            else answer[1] = 6;
            return answer;
    }
}